src/services/Firebase/WriteData/CommunityUtil.tsx

```diff
@@ -0,0 +1,30 @@
+import { doc, updateDoc, addDoc, collection } from 'firebase/firestore';
+import { ICustomCommunity } from 'types/types';
+import UploadFile from '../UploadFile';
+import { customCommunityConverter } from '../Converters/CustomCommunityConverter';
+import { FirebaseAuth, Firestore } from '../FirebaseConfig';
+
+export default async function CreateCustomCommunity(
+  customCommunity: ICustomCommunity,
+  imageFile: File | Blob | undefined
+): Promise<string> {
+  if (FirebaseAuth.currentUser != null) {
+    const CommunityRef = await addDoc(
+      collection(Firestore, 'customCommunities').withConverter(
+        customCommunityConverter
+      ),
+      customCommunity
+    );
+    if (imageFile !== undefined) {
```

Comment:
Instead of adding community document, adding file, updating community document, it would be simpler to first add the image file and then add the community document.

Reply:
need the document id to store the image

Reply:
to make sure the storage path is unique, so when you change the image in the future it will overwrite the existing image